syntax = "proto2";

package Blocksettle.Storage;

message OTPFile
{
   required string   importDateString  = 1;
   required string   otpId             = 2;
   required int64    usedKeysCount     = 3;
   required bytes    nextPrivateKey    = 4;
   required bytes    chainCode         = 5;
   optional bool     encrypted         = 6;
   optional bytes    hash              = 7;
}

message WalletBackupFile
{
   optional string description = 1;
   required string id = 2;

   required bytes seed1 = 3;
   required bytes seed2 = 4;

   required bytes chainCode1 = 5;
   required bytes chainCode2 = 6;

   optional string name = 7;
}

// used for signed transactions
message TXFile
{
   repeated bytes transactions         = 1;
}

message UTXOFile
{
   repeated Transactions txByWallet = 1;

   message Transactions {
      required string wallet = 1;
      repeated bytes  transactions = 2;
   }
}


enum ResponseEntryType
{
   CCTokenEntryType = 1;
   OTPReportEntryType = 2;
   AuthAddressVerificationTransactionType = 3;
   AuthRevokeTransactionType = 4;
   CCDistributionTransactionType = 5;
   AuthAddressVerificationRejectType = 6;
}

message ResponseEntry
{
   required ResponseEntryType type = 1;
   required string            request_id = 2;
   required bytes             data = 3;
}

message ResponseFile
{
   repeated ResponseEntry entries = 1;
}


message CCTokenEntry {
   required string   userId = 1;
   required string   ccToken = 2;
   required string   ccProduct = 3;
   required uint32   amount = 4;
}

message OTPReportEntry {
   required string user_id = 1;
   required string otp_id = 2;
}

message AuthAddressSubmitMixIn {
   required float    fee = 1;
   required uint64   txValue = 2;
   required string   reserveId = 3;
}

message AuthAddressReservationInfo
{
   required string address = 1;
   required string user_name = 2;
   required string reservation_id = 3;
   required string request_id = 4;
}

message AuthAddressVerificationTransaction {
   repeated AuthAddressReservationInfo reservations = 1;
   required bytes verification_transaction = 2;
}

message AuthRevokeTransaction
{
   repeated AuthAddressReservationInfo reservations = 1;
   required bytes revoke_transaction = 2;
}

message CCDistributionTransaction
{
   required string   cc_product = 1;
   repeated string   address = 2;
   required bytes    transaction = 3;
}

message AuthVerificationRejectedMessage
{
   required string user_name = 1;
   required string reservation_id = 2;
}